internal/communicator/ssh/http_proxy.go

```diff
@@ -0,0 +1,168 @@
+package ssh
+
+import (
+	"bufio"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"golang.org/x/net/proxy"
+)
+
+// Dialer implements for SSH over HTTP Proxy.
+type Dialer struct {
+	proxy proxyInfo
+	// forwarding Dialer
+	forward proxy.Dialer
+}
+
+type proxyInfo struct {
+	// HTTP Proxy host or host:port
+	host string
+	// HTTP Proxy scheme
+	scheme string
+	// User name if http proxy needs authentication
+	username string
+	// User password if http proxy needs authentication
+	password string
+	// Whether the HTTP Proxy requires authentication
+	auth bool
+}
```

Comment:
Upon further reading, I think we could omit this new type, and use a `*url.URL` representing the proxy connection instead. While I haven't tried it, so I might be missing something, I think this would be simpler.

Reply:
Fixed